File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        cnt = {}

        for char in s:
            cnt[char] = cnt.get(char, 0) + 1

        for char in t:
            if char not in cnt or cnt[char] == 0:
                return False
            cnt[char] -= 1
        return True

File: Data Structures & Algorithms/anagram-groups/submission-2.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> cnt;
        for(auto s : strs){
            string sortedS = s;
            sort(sortedS.begin(), sortedS.end());
            cnt[sortedS].push_back(s);
        }
        vector<vector<string>> res;
        for(auto s : cnt){
            res.push_back(s.second);
        }
        return res;
    }
};
